### PR TITLE
Add new PCI format support for pcipassthrough

### PR DESCRIPTION
**Pull request by bybai**
Task #3893 

1. added new PCI format support like: 0000:01:00.1
2. updated manpage and doc 

http://10.4.41.1/install/mkvm/xcat-core/docs/build/html/guides/admin-guides/manage_clusters/ppc64le/virtual_machines/manage_vms.html

### CODE REVIEW

File: docs/source/guides/admin-guides/manage_clusters/common/kvm/manage_vm.rst [hunk 1]
```diff
@@ -82,7 +82,24 @@ After the VM object is created, several key attributes need to be specified with
...
+
+     chtab node=vm1 vm.othersettings="devpassthrough:0000:01:00.2" 
+
+    Or: ::
+
+     chtab node=vm1 vm.othersettings="devpassthrough:pci_0000_01_00_2"
+
+    Take assigning SR-IOV VFs to the VM as an example, using ``lspci`` to get VFs PCI and setting the VFs PCI into ``vm`` table: ::
+
+     lspci|grep -i mel
```
Comment (robin2008) [inline docs/source/guides/admin-guides/manage_clusters/common/kvm/manage_vm.rst]: suggest to use `lspci|grep -i "Virtual Function"` as example.

File: docs/source/guides/admin-guides/manage_clusters/common/kvm/manage_vm.rst [hunk 2]
```diff
@@ -82,7 +82,24 @@ After the VM object is created, several key attributes need to be specified with
...
-10. Set **netboot** attribute
+10. (optional)For assigning PCI devices to the VM, set **othersettings** value: ::
+
+     chtab node=vm1 vm.othersettings="devpassthrough:0000:01:00.2" 
+
+    Or: ::
+
+     chtab node=vm1 vm.othersettings="devpassthrough:pci_0000_01_00_2"
+
+    Take assigning SR-IOV VFs to the VM as an example, using ``lspci`` to get VFs PCI and setting the VFs PCI into ``vm`` table: ::
```
Comment (immarvin) [inline docs/source/guides/admin-guides/manage_clusters/common/kvm/manage_vm.rst]: on hypervisor

File: xCAT-server/lib/xcat/plugins/kvm.pm [hunk 3]
```diff
@@ -783,6 +783,13 @@ sub build_xmldesc {
     #prepare the xml hash for pci passthrough
     my @prdevarray;
     foreach my $devname (@passthrudevices) {
+        #This is for SR-IOV vfio
+        #Change vfio format 0000:01:00.2 to pci_0000_01_00_2
+        if ( $devname =~ m/(\w:)+(\w)+.(\w)/ ){
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: should this be ``\d``?

Comment (bybai) [reply xCAT-server/lib/xcat/plugins/kvm.pm]: If the pci is 0000:0e:00.1, should be '\w' .

**Comment by immarvin** [comment]
i am ok with the change

**Comment by robin2008** [comment]
agree to merge

**Pull request merged**